### ISSUE TITLE
ERROR: MethodError: Cannot `convert` an object of type 

### ISSUE THREAD

**Issue by bdeonovic** [dillondaudert/UMAP.jl#34]
```julia
ERROR: MethodError: Cannot `convert` an object of type 
  DataStructures.BinaryHeap{NearestNeighborDescent.KNNGraphs.HeapKNNGraphEdge{Int64{},Float64{}},Base.Order.ReverseOrdering{Base.Order.ForwardOrdering}} to an object of type
  DataStructures.BinaryHeap{NearestNeighborDescent.KNNGraphs.HeapKNNGraphEdge{Int64{},Float64{}},Base.Order.ReverseOrdering}
```

my data looks like this:
```julia
julia> X
10×5503 Array{Float64,2}:
 237.0  237.0  180.0  192.0  192.0  192.0  192.0  822.0  822.0  822.0  561.0  844.0  144.0  564.0  144.0  730.0  …  688.0  834.0  185.0  519.0  376.0  710.0  710.0  171.0  511.0  502.0  511.0  171.0  685.0  166.0  166.0
 141.0  141.0  377.0  375.0  375.0  375.0  375.0  151.0  151.0  151.0  443.0  386.0  394.0  378.0  394.0  391.0     228.0  374.0  304.0  245.0  406.0  240.0  240.0  272.0  232.0  238.0  232.0  300.0  253.0  310.0  310.0      
 516.0  516.0  834.0  834.0  834.0  834.0  834.0  511.0  511.0  193.0  834.0  162.0  834.0  834.0  834.0  834.0     820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0  820.0      
 384.0  384.0  159.0  159.0  159.0  159.0  159.0  334.0  334.0  287.0  132.0  379.0  371.0  371.0  371.0  371.0     180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0  180.0      
 258.0  250.0  411.0  411.0  411.0  411.0  411.0  499.0  499.0  368.0  260.0  339.0  416.0  423.0  416.0  339.0     521.0  577.0  497.0  322.0  167.0  510.0  465.0  571.0  182.0  233.0  182.0  538.0  538.0  513.0  513.0
 381.0  396.0  384.0  384.0  384.0  384.0  384.0   44.0   44.0  421.0  277.0  378.0  378.0  380.0  378.0  378.0  …  247.0  217.0  218.0  398.0  377.0  239.0  262.0  224.0  292.0  303.0  292.0  220.0  220.0  256.0  256.0      
 818.0  818.0  728.0  728.0  728.0  728.0  728.0  210.0  210.0  530.0  530.0  526.0  570.0  170.0  570.0  511.0     317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0  317.0
 366.0  366.0  349.0  349.0  349.0  349.0  349.0  270.0  270.0  213.0  213.0  370.0  372.0  371.0  372.0  370.0     222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0  222.0
   0.0    0.0   37.0   37.0   37.0   37.0   37.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0       0.0    0.0   -7.0    0.0    0.0  -27.0    0.0    0.0  -13.0    0.0  -13.0  -30.0  -30.0  -20.0  -20.0
 -23.0  -23.0  -20.0  -20.0  -20.0  -20.0  -20.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0       0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0
```

and my call looks like

```julia
julia> embedding = umap(X, 2)
```

```julia

julia> versioninfo()
Julia Version 1.5.0
Commit 96786e22cc (2020-08-01 23:44 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: AMD Ryzen 9 3900 12-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, znver1)
Environment:
  JULIA_EDITOR = "C:\Users\bdeon\AppData\Local\Programs\Microsoft VS Code\Code.exe"
  JULIA_NUM_THREADS =
```

**Comment by dillondaudert** [comment]
Looks like this is due to a change in the binary heap constructors in `DataStructures.jl`. I'll look into fixing it (the error arises in NearestNeighborDescent.jl).

**Comment by dillondaudert** [comment]
This turned out to be a bug in DataStructures.jl that was fixed as part of the v0.18.7 release, so try upgrading and let me know if you still have an issue.

**Comment by koalive** [comment]
Thanks, I recently started having the same error when using `umap` on matrices with more than ~5000 columns, but a simple `Pkg.update("DataStructures")` did the trick for me.
